### PR TITLE
Sidebar: WIP Refactor of sidebar CSS

### PR DESCRIPTION
**Pull request by shaunandrews**
WIP

Ok, so this is gonna be a big one, but I thinks a long time coming. The Calypso sidebar has been redesigned recently, but there's some legacy code still lingering around complicating things. The recent redesign overrode existing styles, rather than replaced them.

The goal is to introduce very few visual changes, but reduce and simplify the code where possible. So far, the visual changes are:

- Updating the expandable arrow icon (chevron) with a material icon.
- Move the expandable arrow icon (chevron) for Reader's Lists and Tags to the right.
- Make all menu item's font-weight normal, but expandable items bold.
- "Unstick" the WP Admin and Add New Site menu items from the bottom of the sidebar.

Here's a Before/After for My Site(s) with multiple sites and a single site:

image

### CODE REVIEW

File: client/my-sites/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -584,14 +575,12 @@ export class MySitesSidebar extends Component {
...
-			<Button
-				borderless
-				className="my-sites-sidebar__add-new-site"
-				href={ this.getAddNewSiteUrl() }
-				onClick={ this.focusContent }
-			>
-				<Gridicon icon="add-outline" /> { this.props.translate( 'Add New Site' ) }
-			</Button>
+			<SidebarItem
+				label={ this.props.translate( 'Add new site' ) }
```
Comment (a8ci18n) [inline client/my-sites/sidebar/sidebar.jsx]: Hi! I've found a possible matching string that has already been translated **22** times:
`translate( '+ Add New Site' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/sidebar-CSS-refactor)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/reader/sidebar/index.jsx [hunk 2]
```diff
@@ -228,10 +219,12 @@ export class ReaderSidebar extends React.Component {
 									{ 'sidebar-activity__likes': true }
 								) }
 							>
-								<a href="/activities/likes" onClick={ this.handleReaderSidebarLikeActivityClicked }>
-									<MaterialIcon icon="star_border" />
-									<span className="menu-link-text">{ this.props.translate( 'My Likes' ) }</span>
-								</a>
+								<SidebarItem
+									label={ this.props.translate( 'My likes' ) }
```
Comment (a8ci18n) [inline client/reader/sidebar/index.jsx]: :disappointed: 61 existing translations will be lost with this change.
If the string must change, the following option has already been translated **30** times:
`translate( 'Go to My Likes' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/reader/sidebar/index.jsx [hunk 3]
```diff
@@ -131,108 +130,86 @@ export class ReaderSidebar extends React.Component {
...
-								<a href="/read/search" onClick={ this.handleReaderSidebarSearchClicked }>
-									<MaterialIcon icon="search" />
-									<span className="menu-link-text">{ this.props.translate( 'Search' ) }</span>
-								</a>
-							</li>
+							/>
 
-							<li
+							<SidebarItem
+								label={ this.props.translate( 'My likes' ) }
```
Comment (a8ci18n) [inline client/reader/sidebar/index.jsx]: Hi! I've found a possible matching string that has already been translated **61** times:
`translate( 'My Likes' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/sidebar-CSS-refactor)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by jsnajdr** [comment]
Hi @shaunandrews :wave: How finished or unfinished is this branch? If we rebase it onto the latest `master` and resolve the Git conflicts, is it ready to be reviewed and merged? Or are there any major sub-tasks to be done?

We're migrating our CSS files to webpack and finishing this PR would be a major step towards finishing that migration project.

**Comment by jsnajdr** [comment]
Just rebased this onto latest `master`, resolved conflicts and pushed a few little fixes.

Some issues to resolve:
- this PR maybe breaks or overwrites a sidebar item focus ring 1px glitch that @adamziel fixed in #35138. Adam, can you verify the status of that glitch in this branch?
- another PR that probably breaks is wp-admin link color fix from #34986. @Aurorum can you verify that the new wp-admin link is OK?

The "Add tag" UI in Reader needs some fixing, too:
Screenshot 2019-08-23 at 15 25 52
- there used to be a `Count` of tags as part of the sidebar item, and it's gone now
- the "Add" input should be reset back to empty content (or disappear) after submitted

Cc @diegohaz: this is related to the Sidebar CSS migration.

**Comment by Aurorum** [comment]
> can you verify that the new wp-admin link is OK?

(I'm assuming you refer to the "Add New Site" link which that PR intended to fix)

Afraid it isn't okay :(

Hover state for "Add New Site" on a single site in Nightfall (this isn't exclusive to the "Add New Site" link though, it affects all links in the sidebar):
Screenshot 2019-08-23 at 15 37 26

And on the Ocean scheme:

Screenshot 2019-08-23 at 15 39 39

Also affects the other schemes such as Sunset. For an account with multiple sites, this PR seems fine for the "Add New Site" button on all schemes.

But for the wp-admin link, the icon disappears on hover (scheme: Sunset)

Screenshot 2019-08-23 at 15 43 46

The toggles also aren't changing colour on different schemes, they're just remaining black

**Comment by shaunandrews** [comment]
> there used to be a Count of tags as part of the sidebar item, and it's gone now

I intentionally removed it, as I found it visually complicated and my gut tells me its rarely useful. We can bring it back if there's an objection.

**Comment by jsnajdr** [comment]
> I intentionally removed it, as I found it visually complicated and my gut tells me its rarely useful. We can bring it back if there's an objection.

@shaunandrews No problem, just wanted to confirm that it was the intent 🙂 

I pushed a few more updates with several fixes:

Fixed hover styles in Sunset and Ocean themes, as reported by @Aurorum 

Fixed color of expansion icon: use theme text color instead of black. Buggy state:
Screenshot 2019-08-27 at 10 38 19

Fixed styling of the stats sparkline. Due to absolute positioning and broken CSS styles, it was rendered completely outside the Stats item:
Screenshot 2019-08-27 at 11 00 50
I fixed that by abandoning `position: absolute` and using Flex instead.

Fixed the sidebar in Woo Store, especially the count badges:
Screenshot 2019-08-27 at 11 54 58

As far as I'm concerned, this is now ready to merge 🙂

**Comment by Aurorum** [comment]
Hi @jsnajdr, I spotted two more bugs which this PR would introduce. Firstly, on different colour schemes, the plan name isn't using the right colour variable. On the Ocean scheme: 

Screenshot 2019-08-27 at 13 32 43

Secondly (also on Ocean), when I've selected a tab, I can hover over the icon and it'll change colour. This doesn't occur currently. 

Screenshot 2019-08-27 at 13 33 57

**Comment by jsnajdr** [comment]
Thanks for reporting @Aurorum! Both issues fixed in c7abc02.

**Comment by Aurorum** [comment]
In the Contrast scheme, hovering over the "Plan" tab or a toggle heading renders the plan name/heading title unreadable (doesn't currently happen):

Screenshot 2019-08-27 at 18 57 12
Screenshot 2019-08-27 at 18 56 33

**Comment by jsnajdr** [comment]
Thanks again @Aurorum for thorough testing through all the themes! Another set of CSS fixes for hover and selected states is in 1998940.

File: client/layout/sidebar/style.scss [hunk 4]
```diff
@@ -1,731 +1,206 @@
...
-	@include breakpoint( '<660px' ) {
-		padding: 16px;
-	}
-
-	.gridicon {
-		display: block;
-		float: left;
-		margin-right: 6px;
+	// Some links are marked as "external" and show
+	// and icon to visually indicate they will open
```
Comment (Aurorum) [inline client/layout/sidebar/style.scss]:
```suggestion
	// an icon to visually indicate they will open
```

File: client/layout/sidebar/style.scss [hunk 5]
```diff
@@ -1,731 +1,206 @@
...
+
+// This seems to add a border to the bottom of every
+// .sidebar__menu (whatever that is). I guess to make
+// the menu look better on phones?
+@include breakpoint( '<660px' ) {
+	li.sidebar__menu {
+		border-bottom: 1px solid var( --sidebar-border-color );
 
-		@include breakpoint( '<660px' ) {
-			border-top: none;
```
Comment (Aurorum) [inline client/layout/sidebar/style.scss]: I'm not so sure about getting rid of this, it seems to currently prevent there being a double border on mobile.

Screenshot 2019-08-28 at 08 52 59

But with this branch, there is a double border, and I'm assuming that's not intentional. 

Screenshot 2019-08-28 at 08 52 53

File: client/reader/sidebar/index.jsx [hunk 6]
```diff
@@ -128,111 +127,105 @@ export class ReaderSidebar extends React.Component {
...
```
Comment (Aurorum) [inline client/reader/sidebar/index.jsx]: Minor, but it's currently "My Likes" and that feels like the more consistent option (for example, it's currently "Followed **S**ites".

```suggestion
								label={ this.props.translate( 'My Likes' ) }
```

Screenshot 2019-08-28 at 08 58 20

**Review by Aurorum — COMMENTED** [review]
This PR is testing well, just a few comments though! :)

File: client/layout/sidebar/style.scss (hunk 5 above)
Comment (jsnajdr) [reply client/layout/sidebar/style.scss]: I decided to remove these mobile-only borders. It's hard to make them right in all scenarios, and they are not terribly useful. Other popular mobile apps (looked at Slack and Instagram) don't have borders on menu items and still are great.

**Comment by jsnajdr** [comment]
@Aurorum I implemented all your suggestions:
- fixed a typo in CSS comment
- removed mobile-only borders
- capitalized "My Likes" in Reader sidebar

I also discovered that the `Sidebar` styles were reused (with SASS `import` and `extend`) for the `EditorConfirmationSidebar` in Classic Editor UI:

Screenshot 2019-08-28 at 12 19 08

I had to do a few fixes in 9a4f19f to put it back to order.

File: client/layout/sidebar/style.scss (hunk 5 above)
Comment (Aurorum) [reply client/layout/sidebar/style.scss]: Agreed, it doesn't seem necessary and it still looks good.

**Review by jsnajdr — APPROVED** [review]
OK, time to merge! :tada:

**Review by Aurorum — APPROVED** [review]
I've extensively tested:
- the sidebar at all the different breakpoints 
- the sidebar in Me, My Sites, DevDocs, Shop (though I can't access much there), the classic editor, and the Reader
- the sidebar in all the different colour schemes
- the sidebar for simple sites and Jetpack sites (where there's an external icon)
- the sidebar for both users with one site and multiple sites

Besides the bugs above which have now all been fixed, everything works perfectly as expected!

**Pull request merged**